lib/webauthn/public_key_credential_with_assertion.rb

```diff
@@ -31,5 +32,20 @@ def user_handle
     def raw_user_handle
       response.user_handle
     end
+
+    private
+
+    def appid_extension_output
+      return if client_extension_outputs.nil?
+
+      client_extension_outputs[:appid] || client_extension_outputs['appid']
```

Comment:
> In which case is this appid a symbol?

Yeah maybe it's not needed to make the distinction here, I'm just trying to cover the case where someone could theoretically initialize a `WebAuthn::PublicKeyCredentialWithAssertion` with `client_extension_outputs` being `{ appid: 'my_app_id' }`. 

But in some way I agree that this might not be needed for now as, in reality, the majority of the times this is initialized with `client_extension_outputs: params["clientExtensionResults"]` which will have its keys as strings. Not a strong opinion here.

> I mean, the de-construction of the authenticator's response is always a string so I'm wondering why the distinction here?

Worth mentioning that this is a client extension output – instead of an authenticator extension output – which means that it does not come in the authenticator response 🙂 Your point is still valid though, just wanted to clear that out.

Reply:
Agree with @brauliomartinezlm - this seems defensive programming for an unlikely edge case (probably only existing in the test suite?)

Reply:
Updated in [1d7b5c1](https://github.com/cedarcode/webauthn-ruby/pull/319/commits/1d7b5c13fb3124a978b90b15c2041913867c3a54) 🙂